paddle/framework/attribute.cc

```diff
@@ -74,7 +62,9 @@ Attribute GetAttrValue(const OpDesc::Attr& attr_desc) {
       return val;
     }
     case framework::AttrType::BLOCK: {
-      return GetProgramDesc().mutable_blocks(attr_desc.block_idx());
+      PADDLE_ENFORCE(desc != nullptr,
```

Comment:
Should we make `GetAttrValue` returns the block ID instead of the block itself? If so, we don't need to pass `ProgramDesc*` as a parameter.

Reply:
`GetAttrValue` is only invoked by `CreateOp`. Users will not invoke this method directly. So it is OK to have a `ProgramDesc*` as parameter